### ISSUE TITLE
Feature Request: Metrics in ACA Jobs

### ISSUE THREAD

**Issue by electroma** [microsoft/azure-container-apps#1027]
**Is your feature request related to a problem? Please describe**.  
Regular ACAs have a good set of built-in metrics: replica count, CPU and memory utilization.
I can't find anything like that for ACA Jobs.

**Describe the solution you'd like**.    
Make the following ACA Job metrics available:
* Number of Parallel Job Executions running (similar to Replica Count in regular ACA)
* CPU allocated and utilized (similar to regular ACA)
* Memory allocated and utilized
* Error ratio (container crashes and such)

**Describe alternatives you've considered**.   
I do not see any alternative, and I believe metrics should be published, so the teams can plugin proper monitoring.

**Comment by vinisoto** [comment]
Yes. We are aware of this gap. We expect having these metrics start to flow by Jan 2024. We will update this issue when we have a better ETA.

**Comment by ractando** [comment]
Hey @vinisoto. Is there any ETA on this ?

**Comment by kumarmo-2** [comment]
@vinisoto , Any update on when we can expect this feature to be in GA ?

**Comment by vd84** [comment]
@vinisoto Any update? :D

**Comment by dsczltch** [comment]
Hey @vinisoto. We are also looking forward to this feature very much since we would like to gain more insight into our ACA jobs.

**Comment by vinisoto** [comment]
hi sorry for the lack of updates. We hit some delays on our release. We are targetting for these metrics to be available by the week of 4/22 or sooner.

**Comment by Shalin-AngloAmerican** [comment]
Waiting for this feature please

**Comment by itallackpure** [comment]
Is there an update on progress with this?

**Comment by AdrianProkop** [comment]
Any update on this?

**Comment by pietersap** [comment]
also waiting

**Comment by snavarropino** [comment]
Any news on this?

**Comment by jparta** [comment]
This is a vital enabler for failure monitoring automation. The offering seems incomplete without metrics.

**Comment by k3-yasuda** [comment]
I'm waiting for updates on this feature.

**Comment by nbwdk** [comment]
Insights to metrics are vital for us, in order to do right sizing and consolidations of Dedicated Plans.

**Comment by maverickmetro** [comment]
I see the Otel integration is in preview, will Otel integration get the System metrics for ACA Jobs?

**Comment by vinisoto** [comment]
hi - the metrics UX has been enabled for Jobs:


![image](https://github.com/user-attachments/assets/91512140-daed-4656-ad5f-c48b283302c8)

we currently support:

- Number of Job Executions
- CPU Usage
- Memory Usage

cc: @anthonychu

**Comment by electroma** [comment]
Thank you for the update @vinisoto. 
In the original request I have requested one more important metric - Error Ratio. 
Would it be possible to compute this metric by splitting "Job Executions" on State?

**Comment by rbange** [comment]
@vinisoto when can we except this to be rolled out for all jobs? At least for my subscription its not present yet.

**Comment by anthonychu** [comment]
@rbange The metrics blade should be available everywhere now and should appear as in the screenshot that @vinisoto shared. Could you please check again?

**Comment by rbange** [comment]
@anthonychu Yes the metrics appeared approx 3 days later. However interestingly they are not selectable when trying to access them via scope in the metrics section. There only the regular resources appear, so you basically have to manually go to each job to check them...

Also I can report the same issue as @dinoo. Ram is always maxed out even though they require far less locally and CPU Usage is stuck at around 0.0x nc in min/max and average aggregation. I have a job which runs approx 20 minutes at each full hour and a flat line is absolutely unrealistic.

**Comment by itallackpure** [comment]
> @anthonychu when I check our Job metrics, all 86 jobs report a stable 0.05nc CPU usage and 536.9 MB on average, min and max aggregation.
> 
> Doesn't seem to be right, I wouldn't expect a flatline and the same values for all jobs.

I am seeing the exact same numbers as @dinoo - 0.05nc and 536.9MB average. Something seems wrong...

**Comment by anthonychu** [comment]
Thanks all for reporting. We'll investigate.

**Comment by dani2221** [comment]
Can you clarify what does the metric - Number of job executions mean? I thought it just counted every time the job is started, but as I can see I have more than 7 million executions in the past 30 days on a event driven job that approx. runs 30-50 times a month.

**Comment by vinisoto** [comment]
Hi, thanks for reporting. Some container apps jobs are not displaying the correct runtime value of the CPU and Memory metrics. We are preparing a configuration change and will be rolling it out in the next few days.
 
Regarding Job Executions: This metric displays the current number of job executions in an environment (Consistent with the execution list displayed in the Execution History blade). When a job execution ends, it remains in the environment (currently the last 100 successful and 100 failed executions are kept around). One way to visualize executions in time is to apply a split by Execution Name: 
 
![image](https://github.com/user-attachments/assets/b088603d-7e8d-4ad0-a466-269ae83fb6b6)

 
We are working on two new metrics: Executions Started and Executions Ended to display how many jobs executions start/end at a point in time. Will communicate here once we have a timeline to share.

**Comment by rodyvansambeek** [comment]
> Regarding Job Executions: This metric displays the current number of job executions in an environment (Consistent with the execution list displayed in the Execution History blade). When a job execution ends, it remains in the environment (currently the last 100 successful and 100 failed executions are kept around). One way to visualize executions in time is to apply a split by Execution Name:

This is not working on my Container App Jobs. It runs every hour and I definitely see the runs in the Execution History blade, but nothing in the Metrics section: 

![Image](https://github.com/user-attachments/assets/b14ce933-415a-401a-bd79-e7760bb4a664)


![Image](https://github.com/user-attachments/assets/ead46163-48e1-4d1e-8c6a-344631c2a141)

**Comment by xdawxd** [comment]
Are there any plans to make metrics accessible outside the "Container App Job | Metrics" tab?
I can access them through the ACA Job, but not through Metrics or Dashboard Hub.  
While selecting the scope I can't see any Container App Jobs even though they are in that Resource Group.

**Comment by vinisoto** [comment]
@rodyvansambeek - there was a regression that caused Jobs metrics to stop showing for some customers. We are rolling out a fix which will be fully deployed to all regions by end of this week.

**Comment by vinisoto** [comment]
@xdawxd - we are in the process of fixing both issues: Jobs being available outside of the Job Metrics blade (for example: making Jobs available as a Metrics scope) and being able to create alerts based on Jobs metrics.

We will update here when we have ETAs for both fixes.

**Comment by dss010101** [comment]
does anyone know what the metric are for Job Failed and Job Completed for ACA Jobs?
